packages/flutter_test/lib/src/controller.dart

```diff
@@ -1238,11 +1238,92 @@ abstract class WidgetController {
       double timeStamp = 0.0;
       double lastTimeStamp = timeStamp;
       await sendEventToBinding(testPointer.down(startLocation, timeStamp: Duration(microseconds: timeStamp.round())));
+
       if (initialOffset.distance > 0.0) {
-        await sendEventToBinding(testPointer.move(startLocation + initialOffset, timeStamp: Duration(microseconds: timeStamp.round())));
+        final double xSign = initialOffset.dx.sign;
```

Comment:
I tried removing all changes here and your new test still passes. Is it still necessary?

Reply:
Yes, basically the flingFrom method has an initialOffset paremeter very useful when simulating a swipe/fling after a desplacement, but the current implementation makes initialOffset non funtional, so no displacement is actually performed. [Current behavior of flingFrom.initialOffset ](https://github.com/flutter/flutter/issues/139455#issuecomment-1838323397)

That's why the test passes with the old implementation, because the sheet does not move to the bottom to trigger the edge condition that causes it to close when it should expand again to full height. 

The changes in the controller is to copy the code from dragFrom, that in order to drag a pointer, it has to be divided in smaller distances.

Reply:
Thanks for finding this problem. Can you write a test in `controller_test.dart`? This kind of utility code should also be guarded by tests.

Reply:
Yep. Just precisely test what you found wrong.